api/graphql/filters.js

```diff
@@ -46,65 +44,69 @@ export const personFilter = userId => relation => relation.query(q => {
 })
 
 export const messageFilter = userId => relation => relation.query(q => {
-  q.whereNotIn('user_id', BlockedUser.blockedFor(userId))
+  q.whereNotIn('comments.user_id', BlockedUser.blockedFor(userId))
 })
 
-function filterCommunities (q, idColumn, userId) {
+// Which groups can the user see
+function filterGroups (q, idColumn, userId) {
   // the effect of using `where` like this is to wrap everything within its
   // callback in parentheses -- this is necessary to keep `or` from "leaking"
   // out to the rest of the query
   q.where(inner => {
-    inner
-    .whereIn(idColumn, myCommunityIds(userId))
-    .orWhereIn(idColumn, myNetworkCommunityIds(userId))
-    if (idColumn === 'communities.id') {
-      // XXX: hack to make sure to show public communities on the map when logged in
-      inner.orWhere('communities.is_public', true)
+    // Can see groups you are a member of and their parents
+    const selectIdsForMember = Group.selectIdsForMember(userId)
+    inner.whereIn(idColumn, selectIdsForMember)
+    inner.orWhereIn(idColumn, GroupRelationship.query().select('parent_group_id').whereIn('child_group_id', selectIdsForMember))
+
+    if (idColumn === 'groups.id') {
```

Comment:
Is this still necessary / does it still work?

Reply:
yes